arcaflow_plugin_opensearch/opensearch_plugin.py

```diff
@@ -9,6 +9,49 @@
 from opensearch_schema import ErrorOutput, SuccessOutput, StoreDocumentRequest
 
 
+def convert_to_supported_type(value) -> typing.Dict:
+    type_of_val = type(value)
+    if type_of_val == list:
+        new_list = []
+        for i in value:
+            new_list.append(convert_to_supported_type(i))
+        # A list needs to be of a consistent type or it will
+        # not be indexible into a system like Opensearch
+        return convert_to_homogenous_list(new_list)
+    elif type_of_val == dict:
+        result = {}
+        for k in value:
+            result[convert_to_supported_type(k)] = convert_to_supported_type(value[k])
+        return result
+    elif type_of_val in (float, int, str, bool):
+        return value
+    elif isinstance(type_of_val, type(None)):
+        return str("")
+    else:
+        print("Unknown type", type_of_val, "with val", str(value))
+        return str(value)
+
+
+def convert_to_homogenous_list(input_list: list):
+    # To make all types in list homogeneous, we upconvert them
+    # to the least commom type.
+    # int -> float -> str
+    # bool + None -> str
+    list_type = str()
+    for j in input_list:
+        if type(j) is dict:
+            list_type = dict()
+            break
```

Comment:
I don't think that `dict` is a type to which you can trivially promote scalars (and I suspect that promoting a `list` won't produce the result that you expect):
```
>>> list(map(dict, [1, "2", [3,4], {5:6}]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not iterable
```